cmd/lakectl/cmd/root.go

```diff
@@ -60,6 +61,12 @@ type Configuration struct {
 	} `mapstructure:"credentials"`
 	Server struct {
 		EndpointURL lakefsconfig.OnlyString `mapstructure:"endpoint_url"`
+		Retries     struct {
+			Enabled         bool          `mapstructure:"enabled"`
+			MaxRetries      int           `mapstructure:"max_retries"`
+			MinWaitInterval time.Duration `mapstructure:"min_wait_interval"`
+			MaxWaitInterval time.Duration `mapstructure:"max_wait_interval"`
```

Comment:
These new values should go to the cli reference doc page.
I also suggest to start leaner, maybe just `Enabled` & `MaxRetries` to kick it off? Maybe even just `Enabled`?
I believe the defaults you set would be good enough for almost everyone. I don't feel like we have enough information to determine users need to configure intervals for now.. We can always add more custom config later, removing it is the hard part..

Reply:
I don't see the point of providing fewer configuration options. We lose the flexibility to configure for optimal results, and this way, we'll never get feedback on which settings work for different use cases.

Reply:
Because it's adding scope and friction. Instead of having "retry mechanism" for lakectl, we have "configurable retry mechanism". The gap isn't big coding-wise, I agree. But now I have to ask annoying questions regarding usage, like `MaxRetries` non-positive, `MaxWaitInterval` smaller than `MinWaitInterval`, why choose these values and not `TotalWaitTime`, etc. Without someone requesting something specific we're guessing what they need here, and then changing it might be a breaking change in the cli configuration.